designer/client/src/store/nodeValidationMiddleware.ts

```diff
@@ -4,13 +4,13 @@ import { RootState } from "../reducers";
 import { ActionTypes as UndoActionTypes } from "redux-undo";
 import { debounce } from "lodash";
 import HttpService from "../http/HttpService";
-import { getProcessToDisplay } from "../reducers/selectors/graph";
+import { getProcessToDisplayWithUnsavedName } from "../reducers/selectors/graph";
 
 type ActionType = Action["type"];
 
 const debouncedValidate = debounce(
     (dispatch: ThunkDispatch, getState: () => RootState) =>
-        HttpService.validateProcess(getProcessToDisplay(getState())).then(({ data }) =>
+        HttpService.validateProcess(getProcessToDisplayWithUnsavedName(getState())).then(({ data }) =>
```

Comment:
Its not pretty but its necessary for consistency of limited context validation of scenario properties (where unsaved name is passed differently) with full context validation

Reply:
Okay, so the previous behavior was, that name edit didn't update the id property during validation and now, we make a switch before the validation.  I'm wondering if this name is meaningful, because, now, it implies that, someone needs to have a knowledge, that name, is in fact id.

Reply:
The whole name/id is confusing - these concepts are mixed and in the FE app we mostly use "name" to refer to it (for example see graph selectors). In the BE api its similar. The main thing I'm trying to convey is that we are passing an object using some temporary unsaved fields. Would `getProcessToDisplayWithUnsavedData` be better?

Reply:
I want to just point out, that we switched the ID of the process to the unsaved name, and the user won't know about it without opening this function, but on the other hand, maybe it's just an implementation detail. 

IMO `getProcessToDisplayWithUnsavedName` sounds a little better than `getProcessToDisplayWithUnsavedData`,  so let's leave it as it is :).